### PR TITLE
Added query for composed-task-runner status.

### LINKED ISSUE #5782: Fix the thin task execution page query for CTR status

### PR DESCRIPTION
**Pull request by corneil**
Fixes #5782

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java [hunk 1]
```diff
@@ -44,15 +48,29 @@ public class TaskExecutionThinController {
...
+		this.taskJobService = taskJobService;
 		this.resourceAssembler = new TaskExecutionThinResourceAssembler();
 	}
 
 	@GetMapping(produces = "application/json")
 	@ResponseStatus(HttpStatus.OK)
 	public PagedModel<TaskExecutionThinResource> listTasks(Pageable pageable, PagedResourcesAssembler<AggregateTaskExecution> pagedAssembler) {
-		return pagedAssembler.toModel(explorer.findAll(pageable, true), resourceAssembler);
+		Page<AggregateTaskExecution> page = explorer.findAll(pageable, true);
+		taskJobService.populateComposeTaskRunnerStatus(page.getContent());
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java:63]: Have we determined the cost of running the retrieval of the CTR status for the results set?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java [hunk 2]
```diff
@@ -269,6 +278,31 @@ public Page<JobInstanceExecutions> listJobInstances(String jobName, Pageable pag
...
+				Long id = rs.getLong("TASK_EXECUTION_ID");
+				String ctrStatus = rs.getString("CTR_STATUS");
+				LOG.debug("populateCtrStatus:{}={}", id, ctrStatus);
+				AggregateTaskExecution execution = aggregateTaskExecutionMap.get(id);
+				Assert.notNull(execution, "Expected AggregateTaskExecution for " + id + " from " + ids);
+				updated.incrementAndGet();
+				execution.setCtrTaskStatus(ctrStatus);
+			});
+		}
+		LOG.info("updated {} ctr statuses", updated.get());
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java:303]: Should this be a debug?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java [hunk 3]
```diff
@@ -44,15 +48,29 @@ public class TaskExecutionThinController {
...
+		return pagedAssembler.toModel(page, resourceAssembler);
+	}
+
+	@RequestMapping(value = "", method = RequestMethod.GET, params = "name")
+	@ResponseStatus(HttpStatus.OK)
+	public PagedModel<TaskExecutionThinResource> retrieveTasksByName(@RequestParam("name") String taskName,
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java:69]: Do we have a test to verify the population of the status?

**Review by cppwfs — CHANGES_REQUESTED** [review]
Thanks for the PR, great job!   Had some annoying questions ;-)

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java (hunk 1 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java:63]: I compared before and after timings from `org.springframework.cloud.dataflow.integration.test.tasks.TaskExecutionQueryIT` 
I added index in parent_execution_id and now the difference is hardly noticeable.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java (hunk 2 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java:303]: Yes

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java (hunk 3 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionThinController.java:69]: adding

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 4]
```diff
@@ -313,6 +313,10 @@ public void stopJobExecution(long jobExecutionId, String schemaTarget) throws No
 		logger.info("stopped:{}:{}:status={}", jobExecutionId, schemaTarget, status);
 	}
 
+	@Override
+	public void populateComposeTaskRunnerStatus(Collection<AggregateTaskExecution> taskExecutions) {
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:317]: We probably want to add a test here as well.

**Review by cppwfs — CHANGES_REQUESTED** [review]

**Pull request merged**